python/cudf/cudf/core/dataframe.py

```diff
@@ -217,6 +217,31 @@ def __init__(self, data=None, index=None, columns=None, dtype=None):
                         ),
                     )
                 )
+        elif hasattr(data, "__array_interface__"):
+            arr_interface = data.__array_interface__
+            if (
+                len(arr_interface["shape"]) == 1
+                and len(arr_interface["descr"]) == 1
+            ):
+                new_df = self._from_columns(
+                    [data], index=index, columns=columns
+                )
+            else:
+                new_df = self.from_records(data, index=index, columns=columns)
```

Comment:
What happens if `len(arr_interface["shape"]) > 2`?

Reply:
Which is a 2-d case, and all 2-d cases go through `from_records` (`from_records`) works with 2-d arrays too.

```python
>>> pd.DataFrame.from_records(np.random.randn(10, 2))
          0         1
0 -0.957835 -2.945332
1 -0.651440 -1.191708
2  0.419436  0.906071
3 -1.554972 -0.163832
4 -0.708032 -0.156609
5 -0.592104  0.664202
6  2.507729  0.780785
7 -0.330201 -1.405794
8 -0.734475 -1.065689
9 -0.639492  0.888378
```

Reply:
Sorry what I meant was what if it is 3-D or greater? Do we raise somewhere?

Reply:
Yes we raise an error:

```python
>>> import cupy as cp
>>> a = cp.zeros((2, 3, 4))
>>> a
array([[[0., 0., 0., 0.],
        [0., 0., 0., 0.],
        [0., 0., 0., 0.]],

       [[0., 0., 0., 0.],
        [0., 0., 0., 0.],
        [0., 0., 0., 0.]]])
>>> a.ndim
3
>>> import cudf
>>> cudf.DataFrame(a)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/conda/envs/cudf/lib/python3.7/contextlib.py", line 74, in inner
    return func(*args, **kwds)
  File "/conda/envs/cudf/lib/python3.7/site-packages/cudf/core/dataframe.py", line 223, in __init__
    new_df = self._from_arrays(data, index=index, columns=columns)
  File "/conda/envs/cudf/lib/python3.7/site-packages/cudf/core/dataframe.py", line 4314, in _from_arrays
    f"records dimension expected 1 or 2 but found: {data.ndim}"
ValueError: records dimension expected 1 or 2 but found: 3